specs/chugsplash-manager.md

```diff
@@ -0,0 +1,165 @@
+# ChugSplashManager Speed Spec
+
+## Bundle states
+
+```typescript
+enum ActionType {
+  SET_CODE,
+  SET_STORAGE
+}
+```
+
+```typescript
+enum BundleStatus {
+  EMPTY,
+  PROPOSED,
+  APPROVED,
+  CANCELLED,
+  COMPLETED
+}
+```
+
+```typescript
+struct BundleState {
+  BundleStatus status;
+  bool[] executions;
+  uint256 actionsExecuted;
+  uint256 timeClaimed;
+  address selectedExecutor;
+}
+```
+
+```typescript
+struct Action {
+  string target;
+  ActionType actionType;
+  bytes code;
+}
+```
+
+## Computing a unique bundle ID
+
+```typescript
+const computeBundleId = (
```

Comment:
What makes this unique?  Merkle Roots for identical deployemnts can be the same.  Size and Config could be the same as well.  If someone wants to deploy same thing multiple times, will the config have different target addresses or something?